Source/Synthesis.Solr/ContentSearch/SynthesisSolrFieldNameTranslator.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Linq;
+using System.Reflection;
+using Sitecore;
+using Sitecore.ContentSearch;
+using SolrNet.Schema;
+using Synthesis.ContentSearch;
+
+namespace Synthesis.Solr.ContentSearch
+{
+	public class SynthesisSolrFieldNameTranslator : SynthesisFieldNameTranslator
+	{
+		private readonly SolrSchema _schema;
+		public SynthesisSolrFieldNameTranslator(IProviderSearchContext context, AbstractFieldNameTranslator translator)
+			: base(translator)
+		{
+			//sitecore hides the solr schema behind a private field, we need it to find dynamic fields.
+			var fieldInfo = context.Index.Schema.GetType().GetField("schema", BindingFlags.NonPublic | BindingFlags.Instance);
+			if (fieldInfo != null)
+				_schema = (SolrSchema)fieldInfo.GetValue(context.Index.Schema);
+		}
+
+		public override string GetIndexFieldName(string fieldName)
+		{
+			if (_schema != null && _schema.FindSolrFieldByName(fieldName) != null || _schema.SolrDynamicFields.Any(x => fieldName.EndsWith(x.Name.Substring(1))))
+				return fieldName;
+			//at this point we can't be sure what type the data is in the field, our best bet would be a text field.
+			return AppendSolrText(fieldName);
```

Comment:
we need to do this so that we can do things like:
.Where(x => x["placeholder"] == "content")
also it took a minor hack to get the SolrNet schema which contains the dynamic fields out of the sitecore solr schema.
Note that if it can't find the field or dynamic field it defaults to the fieldname being of _t type